scripts/download_gsp_location_data.py

```diff
@@ -0,0 +1,93 @@
+"""This script downloads the GSP location data from the Neso API and saves it to a CSV file.
+
+This script was used to create the `uk_gsp_locations_20250109.csv` file in the `data` directory.
+"""
+
+import io
+import os
+import tempfile
+import zipfile
+
+import geopandas as gpd
+import pandas as pd
+import requests
+
+SAVE_PATH = "uk_gsp_locations_20250109.csv"
+
+# --- Configuration ---
+GSP_REGIONS_URL = (
+    "https://api.neso.energy/dataset/2810092e-d4b2-472f-b955-d8bea01f9ec0/"
+    "resource/d95e8c1b-9cd9-41dd-aacb-4b53b8c07c20/download/gsp_regions_20250109.zip"
+)
+GSP_REGIONS_GEOJSON_PATH_IN_ZIP = "Proj_27700/GSP_regions_27700_20250109.geojson"
```

Comment:
Could you put in a comment about Proj_2XXX, which is lat long, as we migth want to use that in the future